synapse/handlers/device.py

```diff
@@ -386,6 +415,8 @@ def notify_device_update(self, user_id, device_ids):
             hosts.update(get_domain_from_id(u) for u in users_who_share_room)
             hosts.discard(self.server_name)
 
+        opentracing.set_tag("hosts to update", hosts)
```

Comment:
A tag is set when the information is relevant to the whole span. So here you could say "this span was started in order to update the following hosts". 

Logging is used to mark a single decision made along the way and it will be represented on tracing ui's as a single point in time.

The distinction gets really blurry at times.